From 49905ca7b2e3ed7cd828a1647a3b3335e5e7515d Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Thu, 8 Dec 2011 19:43:24 +0100
Subject: [PATCH 082/285] mythgallery: Improve the list of supported image and
 movie file formats

Since images are read using QImage::load() it is correct to use
QImageReader::supportedImageFormats to specify the list of supported types.

Add *.iso, *.img and *.bin to the list of movie files supoorted by mplayer.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 .../mythgallery/mythgallery/galleryutil.cpp        |   14 ++++++++------
 mythplugins/mythgallery/mythgallery/main.cpp       |   11 ++++++++++-
 2 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/mythplugins/mythgallery/mythgallery/galleryutil.cpp b/mythplugins/mythgallery/mythgallery/galleryutil.cpp
index b4a3b3e..6a838f2 100644
--- a/mythplugins/mythgallery/mythgallery/galleryutil.cpp
+++ b/mythplugins/mythgallery/mythgallery/galleryutil.cpp
@@ -18,6 +18,7 @@
 
 // qt
 #include <QDir>
+#include <QImageReader>
 
 // myth
 #include <mythcontext.h>
@@ -54,13 +55,11 @@ static bool FileDelete(const QFileInfo &file);
 QStringList GalleryUtil::GetImageFilter(void)
 {
     QStringList filt;
-    filt.push_back("*.jpg");
-    filt.push_back("*.jpeg");
-    filt.push_back("*.png");
+
+    Q_FOREACH(QByteArray format, QImageReader::supportedImageFormats())
+        filt.push_back("*." + format);
+
     filt.push_back("*.tif");
-    filt.push_back("*.tiff");
-    filt.push_back("*.bmp");
-    filt.push_back("*.gif");
 
 #ifdef DCRAW_SUPPORT
     filt << DcrawFormats::getFilters();
@@ -73,6 +72,9 @@ QStringList GalleryUtil::GetMovieFilter(void)
 {
     QStringList filt;
     filt.push_back("*.avi");
+    filt.push_back("*.bin");
+    filt.push_back("*.iso");
+    filt.push_back("*.img");
     filt.push_back("*.mpg");
     filt.push_back("*.mp4");
     filt.push_back("*.mpeg");
diff --git a/mythplugins/mythgallery/mythgallery/main.cpp b/mythplugins/mythgallery/mythgallery/main.cpp
index c93f575..ba7d0e0 100644
--- a/mythplugins/mythgallery/mythgallery/main.cpp
+++ b/mythplugins/mythgallery/mythgallery/main.cpp
@@ -4,6 +4,7 @@
 // qt
 #include <QDir>
 #include <QtPlugin>
+#include <QImageReader>
 
 // myth
 #include <mythcontext.h>
@@ -113,9 +114,17 @@ static void setupKeys(void)
     REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
         "MythGallery Media Handler 1/2"), "", "", handleMedia,
         MEDIATYPE_DATA | MEDIATYPE_MIXED, QString::null);
+    QString filt;
+    Q_FOREACH(QByteArray format, QImageReader::supportedImageFormats())
+    {
+        if (filt.isEmpty())
+            filt = format;
+        else
+            filt += "," + format;
+    }
     REG_MEDIA_HANDLER(QT_TRANSLATE_NOOP("MythControls",
         "MythGallery Media Handler 2/2"), "", "", handleMedia,
-        MEDIATYPE_MGALLERY, "gif,jpg,png");
+        MEDIATYPE_MGALLERY, filt);
 }
 
 int mythplugin_init(const char *libversion)
-- 
1.7.9.5

